Articles/Blog/2024-08-TechnicalPropertiesOfSustainableSoftware.md

```diff
@@ -0,0 +1,189 @@
+The Better Scientific Software (BSSw.io) community defines software sustainability as<sup>[1]</sup>:
+
+> The ability of software to continue or evolve.
+> Considerations are often different when viewed from different perspectives (user, developer, manager, funder), but generally relates to resources to maintain or evolve the code, adaptability of the codebase to new uses and new computational platforms.
+
+and as<sup>[2]</sup>:
+
+> Sustainable software means that an existing product remains viable in the future such that it makes sense to continue using, adapting, and expanding its capabilities instead of replacing it with another new or existing product.
+
+and:
+
+> Sustainability is a measure of the cost to maintain and improve a product over its lifetime.
+
+With this latter cost/feasibility focus in mind, we define three different **Levels of Sustainable Software**:
+
+* **Weak Definition**: The same team of developers that originally created the software can continue to add new features and fix bugs in the software at a reasonable cost (i.e., where starting from scratch would be more expensive, even in the long term).
+
+* **Medium Definition**: New developers can reasonably make contributions to the software but these contributions are integrated back by the original development team.
+
+* **Strong Definition**: A different set of developers from the original developers can take over new feature development and bug fixes, and the original developers can go away (and the new team can do so cheaper than starting over from scratch).
+
+The ultimate instance of the "Strong Definition" is  the **Extreme Use Case for Sustainable Software**:
+Your project uses some external software Package X in such a way that it would be very difficult and expensive to rip it out and/or change the code to use something else.
+At some point, the developers of Package X go away, and no one is left in the development community to help extend or support Package X.
+What technical properties of Package X would make it so that you could continue support (with reasonable cost and risk) your own project's usage of Package X, which includes activities like adding some new features, porting to new platforms and compilers, and fixing bugs (i.e., that may be exposed by adding new features and porting to new platforms and upgrading upstream dependencies)?
+
+
+### Key Technical Properties of Sustainable Software
+
+The following technical properties have been widely recognized to lead to software that less expensive to extend and maintain (and is therefore more sustainable).
+
+**Open Source License**:
+Allow the user to change and use the software in any way that is needed for the success of their projects.
+Note that this does not require a fully open-source license in many cases.
+The license just has to allow your project to change the source and use the software as needed for your project.
```

Comment:
I think more clarity on what a "fully open-source license" is, is needed here.  In fact, it might be good to point out that some open source licenses can discourage sustainability, i.e. there are some very good reasons why someone might not want to go anyhere near any GPL'd software.

Reply:
Agreed, picking an open-source license is not obvious and the GPL open-source licenses can actually hinder the usage of software.  This software licensing can be  tricky area, would it make sense to just state something like:

> Picking an software license can be a tricky task and every open-source license is not equivalent.  For example, some open-source licenses like the GNU Public License (GPL), and its variations, can actually restrict the maintenance and usage of the software in many cases.  Software licensing can be is a complex and requires great care.[Reference]

and then we can put in a reference to the article:

* https://bssw.io/items/an-introduction-to-choosing-an-open-source-code-license

?

I don't want to length the article too much on this topic but perhaps the above changes will be enough?

Reply:
Addressed in commit e7d2750f492beb18a852cdee213ca2db61cffe8d?